cvmfs/sync_mediator.cc

```diff
@@ -775,47 +781,45 @@ void SyncMediator::CreateNestedCatalog(SharedPtr<SyncItem> directory) {
 
 void SyncMediator::RemoveNestedCatalog(SharedPtr<SyncItem> directory) {
   const std::string notice =  "Nested catalog at " + directory->GetUnionPath();
-  PrintChangesetNotice(kRemoveCatalog, notice);
+  reporter_->OnRemove(notice, catalog::DirectoryEntry());
 
   if (!params_->dry_run) {
     catalog_manager_->RemoveNestedCatalog(directory->GetRelativePath());
   }
 }
 
+void SyncDiffReporter::OnInit(const history::History::Tag &from_tag,
+                              const history::History::Tag &to_tag) {}
 
-void SyncMediator::PrintChangesetNotice(const ChangesetAction  action,
-                                        const std::string     &extra) const {
-  if (!params_->print_changeset) {
-    ++changed_items_;
-    if (changed_items_ % processing_dot_interval == 0) {
-      LogCvmfs(kLogPublish, kLogStdout | kLogNoLinebreak, ".");
-    }
-    return;
-  }
+void SyncDiffReporter::OnStats(const catalog::DeltaCounters &delta) {}
 
-  const char *action_label = NULL;
-  switch (action) {
-    case kAdd:
-    case kAddCatalog:
-    case kAddHardlinks:
-      action_label = "[add]";
-      break;
-    case kRemove:
-    case kRemoveCatalog:
-      action_label = "[rem]";
-      break;
-    case kTouch:
-      action_label = "[tou]";
-      break;
-    default:
-      assert(false && "unknown sync mediator action");
+void SyncDiffReporter::PrintDots() {
+  if (!print_dots_) return;
+  if (changed_items_ % processing_dot_interval_ == 0) {
+    LogCvmfs(kLogPublish, kLogStdout, ".");
   }
+}
+
+void SyncDiffReporter::InternalAdd(const std::string &path) {
+  const char *action_label = "[add]";
+  LogCvmfs(kLogPublish, kLogStdout, "%s %s", action_label, path.c_str());
```

Comment:
Simply put the `[add]` in the format string, like `LogCvmfs(kLogPublish, kLogStdout, "[add] %s", ...`